### PR TITLE
Remove `CustomProfileField`

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

tl;dr: Talked to @benhalpern and this feature is no longer part of our product plan so we're removing it.

Full context: `CustomProfileField`s were added to allow users to define profile fields unique to their accounts ([issue](https://github.com/forem/forem/issues/2365)). This was originally suggested by a community member to allow for displaying pronouns on their profile. They also added a [PR](https://github.com/forem/forem/pull/7246) to create the model but we never implemented a frontend for it. So when I started working on profile generalization I removed the original code ([PR](https://github.com/forem/forem/pull/9579)) and rebuilt it in a way that fits with our new approach to profiles ([PR]( https://github.com/forem/forem/pull/10202)). However, we also never put a frontend on this new iteration and now decided to get rid of this feature altogether.

## Related Tickets & Documents

Profile generalization 
Closes https://github.com/forem/forem/issues/12160

## QA Instructions, Screenshots, Recordings

There's not much to QA, this feature was never part of our UI. All specs should be green though, otherwise I forgot to remove some.

### UI accessibility concerns?

n/a

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: this never was a public feature.

### CODE REVIEW

File: app/models/profile_field.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class ProfileField < ApplicationRecord
-  include ActsAsProfileField
+  WORD_REGEX = /\b\w+\b/.freeze
```
Comment (citizen428) [inline app/models/profile_field.rb:2]: All the changes in this file are related to inlining the old `ActsAsProfileField` concern that was used to share functionality between `ProfileField` and `CustomProfileField`.

File: db/migrate/20210720042422_drop_custom_profile_fields.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class DropCustomProfileFields < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  def change
```
Comment (citizen428) [inline db/migrate/20210720042422_drop_custom_profile_fields.rb:4]: I didn't bother to make this migration fully reversible by providing a block for `custom_profile_fields`. This is the second time we're removing this feature, I hope it doesn't come back again 😅 

I still tried to keep it somewhat robust, so used `if_exists` which has been around for `drop_table` since 2015 and was added for `remove_index` with Rails 6.1.

File: spec/models/profile_field_spec.rb [hunk 3]
```diff
@@ -1,14 +1,24 @@
 require "rails_helper"
 
 RSpec.describe ProfileField, type: :model do
-  it_behaves_like "a profile field"
```
Comment (citizen428) [inline spec/models/profile_field_spec.rb:4]: The changes to this file are related to inlining the old shared examples.

File: app/models/profile.rb [hunk 4]
```diff
@@ -1,15 +1,10 @@
 class Profile < ApplicationRecord
   belongs_to :user
 
-  validates :data, presence: true
```
Comment (citizen428) [inline app/models/profile.rb:4]: This validation no longer makes sense, since `data` defaults to `{}`. Previously we'd always have a `custom_attributes` key but that is gone now.

File: app/serializers/search/user_serializer.rb [hunk 5]
```diff
@@ -23,11 +22,7 @@ class UserSerializer < ApplicationSerializer
     end
 
     attribute :profile_fields do |user|
-      user.profile.data.except(CUSTOM_ATTRIBUTES).map(&HASH_TRANSFORM) if user.profile
-    end
-
-    attribute :custom_profile_fields do |user|
-      user.profile.data[CUSTOM_ATTRIBUTES].map(&HASH_TRANSFORM) if user.profile
+      user.profile&.data&.map(&HASH_TRANSFORM)
```
Comment (citizen428) [inline app/serializers/search/user_serializer.rb:25]: Rubocop forced me to change from the `if` to the safe navigation operator.

**Review by jgaskins — APPROVED** [review]
Looks good!

Just a couple nitpicks, as is tradition. 🙂

**Comment by citizen428** [comment]
@jgaskins Nitpicks addressed, thanks for those!

**Review by juliannatetreault — APPROVED** [review]
The builds are green and the changes look good to me, so 👍 👍
